core/src/main/java/org/elasticsearch/script/Script.java

```diff
@@ -242,19 +241,14 @@ private Script build(String defaultLang) {
                     throw new IllegalArgumentException("illegal compiler options [" + options + "] specified");
                 }
             } else if (type == ScriptType.STORED) {
-                // Only issue this deprecation warning if we aren't using a template.  Templates during
-                // this deprecation phase must always specify the default template language or they would
-                // possibly pick up a script in a different language as defined by the user under the new
-                // namespace unintentionally.
-                if (lang != null && lang.equals(DEFAULT_TEMPLATE_LANG) == false) {
-                    DEPRECATION_LOGGER.deprecated("specifying the field [" + LANG_PARSE_FIELD.getPreferredName() + "] " +
-                        "for executing " + ScriptType.STORED + " scripts is deprecated; use only the field " +
-                        "[" + ScriptType.STORED.getParseField().getPreferredName() + "] to specify an <id>");
+                if (lang != null) {
+                    throw new IllegalArgumentException(
+                        "illegally specified <lang> for a [" + ScriptType.STORED.getParseField().getPreferredName() + " script");
```

Comment:
For these error messages, I think just explicitly saying "stored scripts" is better now, since the preferred parse field name is just "id".